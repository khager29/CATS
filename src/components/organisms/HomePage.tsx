import React from 'react';
import { Button } from '@greatminds/dp-atomic-ui-lib';

const HomePage = props => {
  const name = props.router.query.name || 'anonymous';

  return (
    <div>
      <h1>
        Hello
        { ' ' }
        {name}
        , from next!
      </h1>

      <p>
        Try adding
        <code>name</code>
        to the query params.
      </p>

      <pre>{JSON.stringify(props, null, ' ')}</pre>
      <Button>Foo</Button>
      { ' ' }
      <Button>Bar</Button>
      { ' ' }
      <Button>
        Qux
      </Button>
    </div>
  );
};

export default HomePage;
