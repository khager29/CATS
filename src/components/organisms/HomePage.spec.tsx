import { render } from '@testing-library/react';
import HomePage from './HomePage';

describe('HomePage', () => {
  it('should render correctly without props', () => {
    const router = {
      query: {},
    };

    const component = render(<HomePage router={router} />).container.firstChild;
    expect(component).toMatchSnapshot();
  });
});
