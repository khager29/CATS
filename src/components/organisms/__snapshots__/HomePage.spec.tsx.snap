// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render correctly without props 1`] = `
.c1 {
  font-size: 1rem;
  font-family: Circular Std;
  font-weight: 400;
  cursor: pointer;
  background: #0B212C;
  color: #FFF;
  border: 0;
  border-radius: 4px;
  padding: 8px 16px;
}

.c1:hover {
  background-color: #4A5D66;
}

.c1:focus:active {
  background: #0B212C;
  -webkit-appearance: none;
  box-shadow: none;
}

.c1:focus {
  outline: none;
  -webkit-appearance: none;
  box-shadow: #FFF 0px 0px 0px 2px,#A6B5BD 0px 0px 0px 5px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 i {
  margin: 0 12px 0 0;
}

.c0 {
  border-radius: 4px;
  padding: 12px 16px;
  color: #0B212C;
  font-family: Circular Std;
  background-color: #FFF;
  border: 1px solid #CCD3D5;
}

<div>
  <h1>
    Hello
     
    anonymous
    , from next!
  </h1>
  <p>
    Try adding
    <code>
      name
    </code>
    to the query params.
  </p>
  <pre>
    {
 "router": {
  "query": {}
 }
}
  </pre>
  <div
    class="c0"
  >
    <button
      class="c1"
    >
      <div
        class="c2"
      >
        Foo
      </div>
    </button>
     
    <button
      class="c1"
    >
      <div
        class="c2"
      >
        Bar
      </div>
    </button>
     
    <button
      class="c1"
    >
      <div
        class="c2"
      >
        Qux
      </div>
    </button>
  </div>
</div>
`;
